src/Services/SchemaPoster.php

```diff
@@ -23,369 +23,382 @@
+        $this->logger->{$log_function}('Files uploaded: {status_code} {reason}', [
+            'status_code' => $response->getStatusCode(),
+            'reason' => $response->getReasonPhrase(),
+        ]);
+        $message = vsprintf('Result: %s Status code: %d - %s', [
+            $log_function == 'error' ? 'NOT UPLOADED' : 'UPLOADED',
+            $response->getStatusCode(),
+            $response->getReasonPhrase(),
+        ]);
+        return [$message];
     }
 
-    return NULL;
-  }
+    /**
+     * Upload schema files to server.
+     *
+     * @param array $schemaFiles
+     *   A key => value paired array of filenames => file_contents.
+     *
+     * @return \Psr\Http\Message\ResponseInterface|null
+     *   A PSR-7 response object from the API call.
+     */
+    public function uploadSchemaFiles(array $schemaFiles): ?ResponseInterface
+    {
+        // Schema upload URL.
+        $uri = new Uri(
+            $this->getClient()
+                ->getEndpoint()
+                ->getSchemaUploadUri()
+        );
+        $this->logger->debug('Upload url: ' . (string)$uri);
+        // Build the files array.
+        $toSend = ['files' => []];
+        foreach ($schemaFiles as $filename => $file_contents) {
+            $this->logger->notice('Encoding file: {filename}', [
+                'filename' => $filename,
+            ]);
+            $toSend['files'][] = [
+                'filename' => $filename,
+                'content' => base64_encode($file_contents),
+            ];
+        }
 
-  /**
-   * Get the schema and config files for posting on the solr server.
-   *
-   * @param string $server_id
-   *   The Search API server id. Typically, `pantheon_solr8`.
-   *
-   * @return array
-   *   Array of key-value pairs: 'filename' => 'file contents'.
-   *
-   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
-   * @throws \Drupal\search_api\SearchApiException
-   * @throws \Drupal\search_api_solr\SearchApiSolrException
-   * @throws \Exception
-   *
-   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
-   */
-  public function getSolrFiles(string $server_id = 'pantheon_solr8') {
-    /** @var \Drupal\search_api\ServerInterface $server */
-    $server = \Drupal::entityTypeManager()
-      ->getStorage('search_api_server')
-      ->load($server_id);
+        // Send the request.
+        $request = new Request(
+            'POST',
+            $uri,
+            [
+                'Accept' => 'application/json',
+                'Content-Type' => 'application/json',
+            ],
+            json_encode($toSend)
+        );
+        $response = $this->getClient()->sendRequest($request);
 
-    if (!$server instanceof EntityInterface) {
-      throw new \Exception(
-        'cannot retrieve the solr server connection settings from the database'
-      );
+        // Parse the response.
+        $log_function = in_array($response->getStatusCode(), [200, 201, 202, 203])
```

Comment:
Useless extra check, `guzzle` will throw an exception for anything non-2xx. Should be converted int `try/catch` block.

Reply:
No, it actually doesn't. if you look at the config HTTP_ERRORS => false will keep It from throwing exceptions for non-200 responses.

Reply:
> No, it actually doesn't. if you look at the config HTTP_ERRORS => false will keep It from throwing exceptions for non-200 responses.

What is the point in resetting this default config? It is safer to follow the default one (and use `try/catch`)